include/dmlc/common.h

```diff
@@ -59,22 +59,12 @@ class OMPException {
 
  public:
   /*!
-   * \brief Parallel OMP blocks should be placed within Run to save exception
+   * \brief should be called in a catch clause to capture the exception
    */
-  template <typename Function, typename... Parameters>
-  void Run(Function f, Parameters... params) {
```

Comment:
Please do not remove the existing facility for exception catching (`Run` and `omp_exc_`). This facility is used in MXNet and TVM, and removing it will necessitate an extensive code refactoring in these projects. (Keep in mind that dmlc-core is not just used by XGBoost but by other projects as well. So backward compatibility is important.) Instead, add `OMP_INIT()`, `OMP_BEGIN()`, and `OMP_END()` as convenience macros that build on the existing facility.

Reply:
@junrushao1994 @szha for visibility